test/unit/BCoWPool.t.sol

```diff
@@ -141,6 +157,142 @@ contract BCoWPool_Unit_EnableTrading is BaseCoWPoolTest {
+    for (uint256 i = 0; i < TOKENS_AMOUNT; i++) {
+      vm.assume(tokens[i] != _token);
+    }
+    _;
+  }
+
+  function test_Revert_NonBoundBuyToken(address _otherToken) public assumeNotBoundToken(_otherToken) {
+    GPv2Order.Data memory order = correctOrder;
+    order.buyToken = IERC20(_otherToken);
+    vm.expectRevert(IBPool.BPool_TokenNotBound.selector);
+    bCoWPool.verify(order);
+  }
+
+  function test_Revert_NonBoundSellToken(address _otherToken) public assumeNotBoundToken(_otherToken) {
+    GPv2Order.Data memory order = correctOrder;
+    order.sellToken = IERC20(_otherToken);
+    vm.expectRevert(IBPool.BPool_TokenNotBound.selector);
+    bCoWPool.verify(order);
+  }
+
+  function test_Revert_LargeDurationOrder(uint256 _timeOffset) public {
+    _timeOffset = bound(_timeOffset, MAX_ORDER_DURATION, type(uint16).max);
+    GPv2Order.Data memory order = correctOrder;
+    order.validTo = uint32(block.timestamp + _timeOffset);
+    vm.expectRevert(IBCoWPool.BCoWPool_OrderValidityTooLong.selector);
+    bCoWPool.verify(order);
+  }
+
+  function test_Revert_NonZeroFee(uint256 _fee) public {
+    _fee = bound(_fee, 1, type(uint256).max);
+    GPv2Order.Data memory order = correctOrder;
+    order.feeAmount = _fee;
+    vm.expectRevert(IBCoWPool.BCoWPool_FeeMustBeZero.selector);
+    bCoWPool.verify(order);
+  }
+
+  function test_Revert_InvalidOrderKind(bytes32 _orderKind) public {
+    vm.assume(_orderKind != GPv2Order.KIND_SELL);
+    GPv2Order.Data memory order = correctOrder;
+    order.kind = _orderKind;
+    vm.expectRevert(IBCoWPool.BCoWPool_InvalidOperation.selector);
+    bCoWPool.verify(order);
+  }
+
+  function test_Revert_InvalidBuyBalanceKind(bytes32 _balanceKind) public {
+    vm.assume(_balanceKind != GPv2Order.BALANCE_ERC20);
+    GPv2Order.Data memory order = correctOrder;
+    order = correctOrder;
+    order.buyTokenBalance = _balanceKind;
+    vm.expectRevert(IBCoWPool.BCoWPool_InvalidBalanceMarker.selector);
+    bCoWPool.verify(order);
+  }
+
+  function test_Revert_InvalidSellBalanceKind(bytes32 _balanceKind) public {
+    vm.assume(_balanceKind != GPv2Order.BALANCE_ERC20);
+    GPv2Order.Data memory order = correctOrder;
+    order = correctOrder;
+    order.sellTokenBalance = _balanceKind;
+    vm.expectRevert(IBCoWPool.BCoWPool_InvalidBalanceMarker.selector);
+    bCoWPool.verify(order);
+  }
+
+  function test_Revert_InsufficientReturn(
+    SwapExactAmountIn_FuzzScenario memory _fuzz,
+    uint256 _offset
+  ) public happyPath(_fuzz) {
+    uint256 _tokenAmountOut = calcOutGivenIn(
+      _fuzz.tokenInBalance, _fuzz.tokenInDenorm, _fuzz.tokenOutBalance, _fuzz.tokenOutDenorm, _fuzz.tokenAmountIn, 0
+    );
+    _offset = bound(_offset, 1, _tokenAmountOut);
+    GPv2Order.Data memory order = correctOrder;
+    order.buyAmount = _fuzz.tokenAmountIn;
+    order.sellAmount = _tokenAmountOut + _offset;
+
+    vm.expectRevert(IBPool.BPool_TokenAmountOutBelowMinOut.selector);
+    bCoWPool.verify(order);
+  }
+}
+
```

Comment:
we do need a `Test_Return_MagicValue` for the happiest of paths, no?

Reply:
it's present on line 362, as part of `isValidSignature` unit tests. `verify()` doesn't return anything, it only reverts if there's something wrong.

I'm not sure it has to be `public` tho, we could save some deployment gas on that

Reply:
ahh perfect then!